agent/mapcall/src/main/java/randoop/instrument/ReplacementFileReader.java

```diff
@@ -0,0 +1,555 @@
+          try {
+            addMethodReplacement(
+                replacementMap, signatureLineMatcher.group(1), signatureLineMatcher.group(2));
+          } catch (ReplacementException e) {
+            throw new ReplacementFileException(e.getMessage(), reader.getLineNumber(), line);
+          }
+        } else {
+          Matcher packageOrClassLineMatcher = PACKAGE_OR_CLASS_LINE.matcher(line);
+          if (packageOrClassLineMatcher.matches()) {
+            try {
+              discoverClassOrPackageReplacements(
+                  replacementMap,
+                  packageOrClassLineMatcher.group(1),
+                  packageOrClassLineMatcher.group(2));
+            } catch (ReplacementException e) {
+              throw new ReplacementFileException(e.getMessage(), reader.getLineNumber(), line);
+            }
+          } else {
+            String msg = "Error in replacement file: bad format";
+            throw new ReplacementFileException(msg, reader.getLineNumber(), line);
+          }
+        }
+      }
+    }
+    return replacementMap;
+  }
+
+  /**
+   * Adds a method replacement described by a pair of method signatures to the replacement map. Uses
+   * reflection to check that the replacement method signature is well-formed and corresponds to a
+   * method that exists.
+   *
+   * <p>See {@link MethodSignature#of(String)} for details on expected format of a method signature.
+   *
+   * @param replacementMap the map from a method to a replacement method to which replacement is
+   *     added
+   * @param originalSignature the signature of the method to be mapped
+   * @param replacementSignature the signature of the method to be mapped
+   * @throws ReplacementException if the replacement signature is badly-formed, the replacement
+   *     class is not found, or the method does not exist
+   */
+  private static void addMethodReplacement(
+      ConcurrentHashMap<MethodSignature, MethodSignature> replacementMap,
+      String originalSignature,
+      String replacementSignature)
+      throws ReplacementException {
+
+    MethodSignature originalDef;
+    try {
+      originalDef = MethodSignature.of(originalSignature);
+    } catch (IllegalArgumentException e) {
+      String msg = String.format("Bad original signature: %s", e.getMessage());
+      throw new ReplacementException(msg);
+    }
+
+    MethodSignature replacementDef;
+    try {
+      replacementDef = MethodSignature.of(replacementSignature);
+    } catch (IllegalArgumentException e) {
+      String msg = String.format("Bad replacement signature: %s", e.getMessage());
+      throw new ReplacementException(msg);
+    }
+
+    Method replacementMethod;
+    try {
+      // check that replacement exists in a way that allows reporting errors
+      replacementMethod = replacementDef.toMethod();
+    } catch (ArgumentClassNotFoundException e) {
+      String msg =
+          String.format("Class not found for replacement argument type %s", e.getMessage());
```

Comment:
If we just use the class name then information is lost about what is wrong with the line.